modin/experimental/core/storage_formats/hdk/query_compiler.py

```diff
@@ -793,3 +795,10 @@ def free(self):
     @property
     def dtypes(self):
         return self._modin_frame.dtypes
+
+
+def _check_int_or_float(op, dtypes):  # noqa: GL08
+    codes = "bBhHiIlLqQpPfd"
+    for t in dtypes:
+        if t.char not in codes:
```

Comment:
```suggestion
    for t in dtypes:
        if not (pandas.api.types.is_integer_dtype(t) or pandas.api.types.is_float_dtype(t)).
```

Reply:
This code fails with RuntimeError: halffloat is not yet supported.

Reply:
I've run this code and it worked perfectly fine:
```python
>>> for t in np.typecodes["All"]:
...     print(t, np.dtype(t))
...     print("\t",pandas.api.types.is_float_dtype(np.dtype(t)))
...     print("\t",pandas.api.types.is_integer_dtype(np.dtype(t)))
...
```
What's the source of this `halffloat` you're using to get the error?

Reply:
This error is raised by HDK. Here is the reproducer - https://github.com/modin-project/modin/pull/5583/files#diff-924b045aa089c05b3a717bc9cea19dc8581e2b8351fcc3c0568a54a27ce8a146R2064

Reply:
Okay, but can we make this piece more human-readable though? Let's define a set of allowed numeric dtypes for HDK somewhere out of this function and then reference it here:
```python
# hdk_on_native/dataframe/utils.py
HDK_ALLOWED_NUMERIC = set(np.typecodes["AllInteger"] + np.typecodes["Float"]) - {np.dtype(np.float16).char}

def check_if_allowed_numeric(dtypes):
    ...
```

Reply:
Done.